beacon-chain/core/blocks/block_operations.go

```diff
@@ -43,18 +46,38 @@ func VerifyProposerSignature(
 //   state.eth1_data_votes.append(body.eth1_data)
 //   if state.eth1_data_votes.count(body.eth1_data) * 2 > SLOTS_PER_ETH1_VOTING_PERIOD:
 //     state.latest_eth1_data = body.eth1_data
-func ProcessEth1DataInBlock(beaconState *pb.BeaconState, block *pb.BeaconBlock) *pb.BeaconState {
+func ProcessEth1DataInBlock(beaconState *pb.BeaconState, block *pb.BeaconBlock) (*pb.BeaconState, error) {
 	beaconState.Eth1DataVotes = append(beaconState.Eth1DataVotes, block.Body.Eth1Data)
-	numVotes := uint64(0)
-	for _, vote := range beaconState.Eth1DataVotes {
-		if proto.Equal(vote, block.Body.Eth1Data) {
-			numVotes++
+
+	voteCount, err := eth1DataCache.Eth1DataVote(block.Body.Eth1Data.DepositRoot)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve eth1 data vote cache: %v", err)
+	}
+
+	if voteCount == 0 {
+		for _, vote := range beaconState.Eth1DataVotes {
+			if proto.Equal(vote, block.Body.Eth1Data) {
+				voteCount++
+			}
+		}
+		if err := eth1DataCache.AddEth1DataVote(&cache.Eth1DataVote{
+			DepositRoot: block.Body.Eth1Data.DepositRoot,
+			VoteCount:   voteCount,
+		}); err != nil {
+			return nil, fmt.Errorf("could not save eth1 data vote cache: %v", err)
+		}
+	} else {
+		voteCount, err = eth1DataCache.IncrementEth1DataVote(block.Body.Eth1Data.DepositRoot)
```

Comment:
anyway to avoid retrieving the vote count again from the cache here ? We do it once above, would be great if we could reuse that.